File: server/router/testcase/testcase.go
package testcase

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/yunfei07/lemon/server/api/v1"
	"github.com/yunfei07/lemon/server/middleware"
)

type TestCaseRouter struct {}

func (tc *TestCaseRouter) InitTestCaseRouter(Router *gin.RouterGroup){
	testcaseRouter := Router.Group("testcase").Use(middleware.OperationRecord())
	testcaseRouterWithoutRecord := Router.Group("testcase")
	var testcaseApi = v1.ApiGroupApp.TestCaseApiGroup.TestcaseApi
	{
		testcaseRouter.POST("createTestCase",testcaseApi.CreateTestCase) // 新建
		testcaseRouter.PUT("updateTestCase",testcaseApi.UpdateTestCase) // 更新
		testcaseRouter.DELETE("deleteTestCaseByIds",testcaseApi.DeleteTestCaseByIds) // 通过ID删除
		testcaseRouter.DELETE("deleteTestCase",testcaseApi.DeleteTestCase)
		testcaseRouter.DELETE("deleteTestStepByIds",testcaseApi.DeleteTestStepByIds) // 通过ID删除步骤
		testcaseRouter.DELETE("deleteTestStep",testcaseApi.DeleteTestStep) // 删除步骤
	}
	{
		testcaseRouterWithoutRecord.GET("getTestCase",testcaseApi.FindTestCase) // 通过ID获取
		testcaseRouterWithoutRecord.GET("runTestCase",testcaseApi.RunTestCase) // 运行case
		testcaseRouterWithoutRecord.GET("getTestCaseList",testcaseApi.GetTestCaseList) // 获取caseList
	}
}

File: web/src/api/testcase.js
import service from '@/utils/request'

export const createTestcase = (data) => {
  return service({
    url: '/testcase/createTestCase',
    method: 'post',
    data,
  })
}

export const updateTestcase = (data) => {
  return service({
    url: '/testcase/updateTestCase',
    method: 'put',
    data: data,
  })
}

export const getTestCaseList = (params) => {
  return service({
    url: '/testcase/getTestCaseList',
    method: 'get',
    params,
  })
}

export const getTestCase = (params) => {
  return service({
    url: '/testcase/getTestCase',
    method: 'get',
    params,
  })
}

export const runTestCase = (params) => {
  return service({
    url: '/testcase/runTestCase',
    method: 'get',
    params,
  })
}

export const debugTestCase = (params) => {
  return service({
    url: '/testcase/debug',
    method: 'get',
    params,
  })
}

export const deleteTestCase = (data) => {
  return service({
    url: '/testcase/deleteTestCase',
    method: 'delete',
    data,
  })
}

export const deleteTestCaseByIds = (data) => {
  return service({
    url: '/testcase/deleteTestCaseByIds',
    method: 'delete',
    data,
  })
}

export const deleteTestStep = (data) => {
  return service({
    url: '/testcase/deleteTestStep',
    method: 'delete',
    data,
  })
}

export const deleteTestStepByIds = (data) => {
  return service({
    url: '/testcase/deleteTestStepByIds',
    method: 'delete',
    data,
  })
}
